include/rmm/device_buffer.hpp

```diff
@@ -32,74 +32,79 @@ namespace rmm {
  * device_buffer buff_copy(buff);
+ * 
+ * // deep copies `buff` into a new device buffer using the specified stream
+ * device_buffer buff_copy(buff, stream);
  *
  * // shallow copies `buff` into a new device_buffer, `buff` is now empty
  * device_buffer buff_move(std::move(buff));
  *
  * // Default construction. Buffer is empty
  * device_buffer buff_default{};
  *
- * // If the requested sized is larger than the current
- * // size, resizes allocation to the new size and deep copies any previous
- * // contents. Otherwise, simply updates the value of `size()` to the newly
- * // requested size without any allocations or copies
- * buff_default.resize(100);
+ * // If the requested size is larger than the current size, resizes allocation
+ * // to the new size and deep copies any previous contents. Otherwise, simply
+ * // updates the value of `size()` to the newly requested size without any
+ * // allocations or copies. Uses the optionally specified stream or the default
+ * // stream if none specified.
+ * buff_default.resize(100, stream);
  *```
  *---------------------------------------------------------------------------**/
-
 class device_buffer {
  public:
-  /**---------------------------------------------------------------------------*
+  /**--------------------------------------------------------------------------*
    * @brief Constructs an empty `device_buffer` of size 0
-   *---------------------------------------------------------------------------**/
+   *-------------------------------------------------------------------------**/
   device_buffer() = default;
 
-  /**---------------------------------------------------------------------------*
-   * @brief Constructs a new device buffer of `size` unitialized bytes
+  /**--------------------------------------------------------------------------*
+   * @brief Constructs a new device buffer of `size` uninitialized bytes
    *
-   * @throws std::bad_alloc If creating the new allocation fails
+   * @throws std::bad_alloc If allocation fails.
    *
-   * @param size Size in bytes to allocate in device memory
+   * @param size Size in bytes to allocate in device memory.
    * @param stream CUDA stream on which memory may be allocated if the memory
-   * resource supports streams, otherwise the default stream is used.
-   * @param mr Memory resource to use for the device memory allocation
-   *---------------------------------------------------------------------------**/
+   * resource supports streams.
+   * @param mr Memory resource to use for the device memory allocation.
+   *-------------------------------------------------------------------------**/
   explicit device_buffer(
       std::size_t size, cudaStream_t stream = 0,
       mr::device_memory_resource* mr = mr::get_default_resource())
       : _size{size}, _capacity{size}, _stream{stream}, _mr{mr} {
-    _data = _mr->allocate(size, stream);
+    _data = _mr->allocate(size, _stream);
```

Comment:
Just a style thing, but ever since I implemented the memoized null count in column I've taken to preferring to use public accessors internally whenever possible. Just in case you want to put some extra logic in there in the future. 
```suggestion
    _data = _mr->allocate(size, stream());
```

Reply:
I had that change but because the name stream is already a parameter it wouldn't compile as-is. Seemed uglier to put this-> in front of every use of stream().

Reply:
Fixed.